### PR TITLE
bugfix/host-videos-under-more-scenarios

### PR DESCRIPTION
**Pull request by JacksonMaxfield**
### Description of Changes

_Include a description of the proposed changes._

Major goal of this work is to host the video file provided to us from scraper under more scenarios.

However in addition to that, I added an option to the clean database function that allows the skipping of cleaning the index collections because those take a long time to fully delete. Additionally, I fix a bug I found while running the pipeline for king county which was that king county has some files that have names longer than 255 characters which Ubuntu doesn't like.

I can't seem to figure out why these changes will work in synchronous mode but not in parallel...

### CODE REVIEW

File: cdp_backend/pipeline/event_gather_pipeline.py [hunk 1]
```diff
@@ -301,33 +312,69 @@ def convert_video_to_mp4_and_upload(
...
-        # Return converted local mp4 filepath
-        return mp4_filepath
+        # Create fs to generate hosted media URL
+        hosted_video_media_url = fs_functions.get_open_url_for_gcs_file(
+            credentials_file=credentials_file,
+            uri=hosted_video_uri,
+        )
 
-    return video_filepath
+    return video_filepath, hosted_video_media_url
```
Comment (dphoria) [inline cdp_backend/pipeline/event_gather_pipeline.py:377]: Just wondering/trying to think in my head, does every code path to this statement include definition of the `hosted_video_media_url` somewhere in its path? In particular, I'm wondering about the `# Convert to mp4 if necessary` `if...else` blob.

**Review by dphoria — APPROVED** [review]
That one question/thought I commented earlier is it for me.

File: cdp_backend/pipeline/event_gather_pipeline.py (hunk 1 above)
Comment (JacksonMaxfield) [reply cdp_backend/pipeline/event_gather_pipeline.py:377]: Yea I had to check this a number of times but it does exist in all cases.

Also, this is exactly why we use codecov! https://app.codecov.io/gh/CouncilDataProject/cdp-backend/compare/138/diff

image

Our tests hit every line of the code.

Comment (dphoria) [reply cdp_backend/pipeline/event_gather_pipeline.py:377]: Whoa this is pretty awesome. Cool!

Comment (JacksonMaxfield) [reply cdp_backend/pipeline/event_gather_pipeline.py:377]: Correction, looks like I am missing a single line 348, but that is simply setting `cdp_will_host` which we know will enable the hosting so no worries there.

Comment (dphoria) [reply cdp_backend/pipeline/event_gather_pipeline.py:377]: OK got it I follow

**Pull request merged**